packages/wasm/src/index.ts

```diff
@@ -40,26 +89,66 @@ export function wasm(options: RollupWasmOptions = {}): Plugin {
           }
         }
 
-        if (imports && !sync) {
-          return WebAssembly.instantiate(buf, imports)
-        } else if (!imports && !sync) {
-          return WebAssembly.compile(buf)
-        } else {
+        if(sync) {
           var mod = new WebAssembly.Module(buf)
           return imports ? new WebAssembly.Instance(mod, imports) : mod
+        } else {
+          return _instantiateOrCompile(buf, imports, false)
         }
       }
     `.trim(),
 
     transform(code, id) {
       if (code && /\.wasm$/.test(id)) {
-        const src = Buffer.from(code, 'binary').toString('base64');
-        const sync = syncFiles.indexOf(id) !== -1;
-        return `export default function(imports){return _loadWasmModule(${+sync}, '${src}', imports)}`;
+        const isSync = syncFiles.indexOf(id) !== -1;
+        const filepath = copies[id] ? `'${copies[id]}'` : null;
+        let src;
+
+        if (filepath === null) {
+          assert(!isSync, 'non-inlined files can not be `sync`.');
```

Comment:
we really don't use `assert` in plugins. instead, we'll use `this.error` or `this.warn`